superduperdb/components/component.py

```diff
@@ -45,6 +44,23 @@ def __post_init__(self, artifacts):
         if not self.identifier:
             raise ValueError('identifier cannot be empty or None')
 
+    def init(self, db=None):
+        from superduperdb.base.document import Document
+        from superduperdb.components.datatype import Encodable
+
+        for f in dc.fields(self):
+            item = getattr(self, f.name)
+            if isinstance(item, Component):
```

Comment:
so this is ensure initialise for field dependent components

Reply:
Yes.